Tweak outreach map

PsyCommando opened the pull request:
## Description of changes
Remake the outreach map to be a bit more up to date and interesting.

## Changelog
:cl:
tweak: Tweaked a lot of things.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

PsyCommando committed:
recycler wip

forgot dust

recycler

PsyCommando commented:
Gotta clean the trash in this PR

PsyCommando commented:
Alright, so it passes CI now.

NataKilar commented on maps/kleibkhar/kleibkhar_exoplanet.dm:
```diff
@@ -1,51 +1,87 @@
-	water_color = COLOR_BLUE_GRAY
-	crust_strata = /decl/strata/base_planet
+/datum/map/kleibkhar/build_exoplanets()
+	log_debug("MAP: building exoplanets!") //#REMOVEME
```
Debug print

NataKilar commented on maps/kleibkhar/kleibkhar_exoplanet.dm:
```diff
@@ -99,21 +135,73 @@
+/decl/strata/kleibkhar
+	default_strata_candidate = TRUE
+
+/decl/strata/kleibkhar/subterrane
```
Subterranean or subterrane? I know both are words.

PsyCommando replied:
Eh?

NataKilar commented on maps/outreach/outreach_defines.dm:
```diff
@@ -1,33 +1,31 @@
-	starting_money = 5000
-	department_money = 0
-	salary_modifier = 0.2
+	//#REMOVEME: Shouldn't get commited!
```
Should this be removed?

PsyCommando replied:
Oh, yep.

NataKilar commented on maps/outreach/outreach_exoplanet.dm:
```diff
@@ -1,69 +1,224 @@
-	water_color = COLOR_BOTTLE_GREEN
-	crust_strata = /decl/strata/sedimentary
+	//Sector Access
+	free_landing          = FALSE
```
This needs to be TRUE, otherwise nothing can land and take off from the surface.

PsyCommando replied:
Speaking of that. I really don't think we should let shuttles land anywhere on the planet, but on docking areas?

Because the shuttle code still lets you land on any z-levels of the planet, and pretty much anywhere without an area protection thing. 
That in turn is going to mess with the base turfs when the thing takes off. And I just can't guarantee it won't cause problems or end up being used for griefing.

For reference, landing on a transition edge is going to permanently fuck the map. The underground turfs ended up being less trouble being simulated turfs for allowing laying down cable and etc,. So a shuttle landing there is very bad..

I can turn it back on, but like, can we at least add support for docking ports, so we don't have to deal with shuttle damage again? It's really hard to recover anything that a shuttle destroys.

NataKilar commented on maps/outreach/outreach_overmap.dm:
```diff
@@ -0,0 +1,50 @@
+	event_areas = 10
+	map_size_x  = 50
+	map_size_y  = 50
+	var/map_file// = "maps/outreach/outreach-overmap.dmm"
```
Should this be commented out?

PsyCommando replied:
Oh, that actually reminds me I didn't test the overmap -_-
But yeah that's fine. We really shouldn't need to pre-generate the overmap.

NataKilar commented on maps/outreach/outreach_overmap.dm:
```diff
@@ -0,0 +1,50 @@
+/////////////////////////////////////////////////////////////////////////////////
+// Supply Shuttle Handling
+/////////////////////////////////////////////////////////////////////////////////
+#if 0
```
Make this an actual define or remove it

NataKilar commented on maps/outreach/outreach_weather.dm:
```diff
@@ -0,0 +1,83 @@
+/decl/state/weather/salt_flurry
+	name       = "Salt Flurry"
+	icon_state = "snowfall_light"
+	descriptor = "The wind is blowing Chlorine salts everywhere."
```
Chlorine -> chlorine

NataKilar commented on maps/outreach/turfs/outreach_turf_flooded.dm:
```diff
@@ -0,0 +1,12 @@
+/turf/simulated/floor/asteroid/outreach/water/chlorine/Initialize(ml, floortype)
+	. = ..()
+	make_flooded(TRUE)
+	add_fluid(/decl/material/gas/chlorine, FLUID_MAX_DEPTH)
```
Shouldn't this be hydrochloric acid? Liquid chlorine would rapidly boil

NataKilar commented on mods/persistence/modules/overmap/_overmap.dm:
```diff
@@ -1,30 +1,30 @@
-// Complete override since Persistence hotloads it maps, and most of these functions must be delayed.
```
If this has been depreciated you can delete the file

PsyCommando replied:
kk

NataKilar commented on mods/persistence/modules/overmap/exoplanets/_exoplanet.dm:
```diff
@@ -1,8 +1,10 @@
+//#TODO: Gotta fix this so we just run the exoplanet's gen code on startup instead to properly update areas and stuff for unsaved turfs.
```
Has this been fixed?

PsyCommando replied:
Nope. I just added a todo for it.

PsyCommando replied:
Nope. This will require work upstream. I think loaf mentioned he'd fix planet so they init at the same time as everything else. But it probably will take a while.

NataKilar commented on code/__defines/overmap.dm:
```diff
@@ -29,3 +29,11 @@
+#define ZLEVEL_CONTACT    BITFLAG(2)
+#define ZLEVEL_PLAYER     BITFLAG(3)
+#define ZLEVEL_SEALED     BITFLAG(4)
+#define ZLEVEL_PERSISTENT BITFLAG(5) //If set the z-level will be added to the saved level list
```
The last two defines should be moved into the Persistence modpack

PsyCommando replied:
Can't do that. It's used in the map. It would fail unit test on the map if the modpack wasn't loaded.
Those probably should get moved upstream too. I was going to when getting the lateral z stuff upstream, but that's gonna be a while off..

NataKilar commented on code/game/machinery/doors/airlock_control.dm:
```diff
@@ -126,22 +126,7 @@
-	. = ..()
-	update_icon()
-
-/obj/machinery/airlock_sensor/examine(mob/user, distance, infix, suffix)
```
Was this examine text replaced somewhere?

PsyCommando replied:
Probably something that slipped during the merge..

NataKilar commented on code/game/objects/effects/effect_system.dm:
```diff
@@ -69,7 +69,8 @@ steam.start() -- spawns the effect
 	for(i=0, i<src.number, i++)
-		addtimer(CALLBACK(src, /datum/effect/effect/system/proc/spread, i), 0)
+		spread(i)
+		//addtimer(CALLBACK(src, /datum/effect/effect/system/proc/spread, i), 0)
```
Why was this changed? I believe the 0 delay callback is intentional.

PsyCommando replied:
It was removed because it caused massive lag when you had a few geysers on the map, and a lot of timer runtimes. Plus, spread already is async anyways?
I actually rewrote that effect for geysers using particles though.

PsyCommando replied:
I'll revert that

NataKilar commented on code/modules/organs/internal/lungs.dm:
```diff
@@ -80,6 +80,8 @@
 		exhale_type =         species.exhale_type  || /decl/material/gas/carbon_dioxide
 	else
+		//Species-less lungs should be illegal
+		log_debug("Got null species")
```
Might want to make this debug message more specific

PsyCommando replied:
Yeah, that shouldn't be in here. I was debugging why mobs would runtime when breathing.

NataKilar commented on code/modules/overmap/sectors.dm:
```diff
@@ -60,6 +60,7 @@ var/global/list/known_overmap_sectors
 	. = ..()
 
 /obj/effect/overmap/visitable/Initialize()
+	log_debug("[src]([x],[y],[z]) creating sector.") //#REMOVEME
```
Left over debug lines?

NataKilar commented on code/modules/overmap/sectors.dm:
```diff
@@ -125,6 +126,7 @@ var/global/list/known_overmap_sectors
 	return get_filtered_areas(list(/proc/area_belongs_to_zlevels = map_z))
 
 /obj/effect/overmap/visitable/proc/find_z_levels()
+	log_debug("[src]([x],[y],[z]) finding z levels. Current z is [z].") //#REMOVEME
```
Ditto

NataKilar commented on code/modules/overmap/ships/landable.dm:
```diff
@@ -116,12 +116,14 @@
 		landmark.shuttle_name = shuttle
 		LAZYDISTINCTADD(initial_generic_waypoints, landmark.landmark_tag) // this is us being user-friendly: it means we register it properly regardless of whether the mapper put the tag in initial_restricted_waypoints
 
+	/* //#OUTREACH: Since we delete those right when creating them, might as well just comment this out
```
This is fine, but you might as well delete the part of the code that actually deletes them as well.

PsyCommando replied:
Just for reference, that was to fix the hard deletes. 
I'd suggest maybe doing that in another PR? Because this one is extremely bloated as it is.

PsyCommando committed:
kleibkhar map

PsyCommando committed:
kleibkhar stuff

PsyCommando committed:
Update designs_misc_circuits.dm

PsyCommando committed:
geothermal

PsyCommando committed:
Overmap handling changes

Stuff modified from base neb overmap.

PsyCommando committed:
Added vent pump variants

new vents subclasses

PsyCommando committed:
materials

PsyCommando committed:
add test skipping

PsyCommando committed:
new sound effects

add cryopod loop

PsyCommando committed:
map change

PsyCommando committed:
removed old overmap override

PsyCommando committed:
Outreach Map

turfs

PsyCommando committed:
outreach level stuff

PsyCommando committed:
Update chargen_areas.dm

PsyCommando committed:
shuttle template

PsyCommando committed:
icons

PsyCommando committed:
Update folding.dm

PsyCommando committed:
Minor decoration + furnishing

* Added recycler on disposals level.
* Furnished hangar.
* Decals.
* Mining drills stuff.

Misc map changes

* Painted some walls
* Fixed missing power cable to brig
* Set brig timers id tags
* Furnished mining department.
* Added dirt
* Placed down recycler. It's currently busted however.

PsyCommando committed:
Update outreach-1.dmm

PsyCommando committed:
recycler fixed

fixed recycler linter warning

PsyCommando committed:
Added a lot of dirt

* Painted dirt over the map.
* Added some clutter.
* Decorated mining levels.

PsyCommando committed:
fixed some decals missing icons

PsyCommando committed:
merge fixes

PsyCommando committed:
Moved outreach to map_template

Outreach is initialized via map template, which simplifies things.

PsyCommando committed:
Cleaned up turf types.

PsyCommando committed:
linter hates comments in lists

PsyCommando committed:
Fix compilation after merge

* Fixed a bunch of things so the map compiles and works.
- Currently, the mining levels don't connect to the main z-stack because some of the init order changes aren't in yet. Waiting for exoplanet gen rework.

PsyCommando committed:
Mining use midnight rollover checked timeofday

Changed world.timeofday with REALTIMEOFDAY to ensure midnight rollover doesn't make mining regen freeze for an entire day if  it's still running past midnight.

PsyCommando committed:
Update recycler.dm

PsyCommando committed:
Compile fix outreach

PsyCommando committed:
Properly use constants for network name

PsyCommando committed:
Use constants for protected area definition

PsyCommando committed:
Update outreach_areas.dm

PsyCommando committed:
fix poster offset

PsyCommando committed:
fix crema switch

PsyCommando committed:
Fix really shady override of sequential ids

Sequential ids were being completely overridden to exclusively assign guid to everything. This caused problem for things that just want to have a name or text numbered in order.
Like mices, poster numbers, etc..
Made a lot of other things use the new guid generator when appropriate.

PsyCommando committed:
move network stuff to own file

PsyCommando committed:
Update recycler.dm

PsyCommando committed:
guids include in neb

PsyCommando committed:
Moved helper code to inside modpack

Upstream removed stacks, so...

PsyCommando committed:
remove old strata var

PsyCommando committed:
docking controller nuked

Docking controller being their usual bugged PoS selves are crashing over duplicate program instances for their convoluted control system. So I just removed the supply shuttle stuff from the map because I'm kinda sick of having to constantly deal with it breaking everything.

PsyCommando committed:
Make turbolift generate last

PsyCommando committed:
properly offset wall recharger

PsyCommando committed:
Properly offset wall dispensers

PsyCommando committed:
comment sequential guid gen

PsyCommando committed:
godmode check on fire acts

PsyCommando committed:
Don't save mimic edges

PsyCommando committed:
Update outreach_elevators.dm

PsyCommando committed:
Merge remote-tracking branch 'upstream/dev' into tweak-outreach-map

PsyCommando committed:
Update outreach-4.dmm

PsyCommando committed:
satisfy mapping gods

PsyCommando committed:
Update outreach_south-2.dmm

PsyCommando committed:
added missing unit test overrides

PsyCommando committed:
fixes missing species

PsyCommando committed:
fix duplicate catwalk spawning

PsyCommando committed:
formatting changes

PsyCommando committed:
Get area tests to behave

PsyCommando committed:
Fix missing scrubber

PsyCommando committed:
Some area fixes

PsyCommando committed:
Fix list error during unit tests

PsyCommando committed:
Fix internet uplink templates

PsyCommando committed:
Update outreach_test.dm

PsyCommando committed:
Test

PsyCommando committed:
Merge remote-tracking branch 'upstream/dev' into tweak-outreach-map

NataKilar commented on code/controllers/subsystems/mapping.dm:
```diff
@@ -109,6 +107,10 @@ SUBSYSTEM_DEF(mapping)
 		level.setup_level_data()
 
+	// Generate turbolifts last!!!
+	for(var/obj/abstract/turbolift_spawner/turbolift as anything in turbolifts_to_initialize)
```
Is this something specific for Persistence or could it go upstream eventually?

PsyCommando replied:
I think it went upstream already?

PsyCommando replied:
It's not unique to outreach, any dynamically loaded maps like away_sites would have that issue if they had elevators.

NataKilar commented on code/game/objects/items/stacks/dust.dm:
```diff
@@ -0,0 +1,72 @@
+	use(amount)
+
+// Dust override
+/decl/material
```
Instinct tells me this should be in the base material file, but the Neb maints can decide once/if it goes upstream

NataKilar commented on code/modules/fabrication/designs/imprinter/designs_misc_circuits.dm:
```diff
@@ -485,6 +485,11 @@
+	path = /obj/item/stock_parts/circuitboard/geothermal
+
+/datum/fabricator_recipe/imprinter/circuit/radiocarbon_spectrometer
+	path = /obj/item/stock_parts/circuitboard/radiocarbon_spectrometer
```
New line here if you get the chance

NataKilar commented on code/unit_tests/map_tests.dm:
```diff
@@ -633,6 +634,9 @@
 	if(!isStationLevel(source_turf.z))
 		return TRUE
 
+	if(locate(/obj/abstract/landmark/skip_test) in source_turf)
```
Is this the only test this should be checked?

PsyCommando replied:
For now yes. But it's a work in progress tool for convenience.

NataKilar commented on maps/outreach/outreach_machinery.dm:
```diff
@@ -0,0 +1,219 @@
+////////////////////////////////////////////////////////////////////////
+// Defense Turret
+////////////////////////////////////////////////////////////////////////
+/obj/machinery/porta_turret/stationary/stun
```
Making a note that we'll need to replace these with the new turrets once they're in

NataKilar reviewed (approved):
Further map adjustments can be made in future PRs. This seems fine.
